### PR TITLE
PIM-5737: add spin

### PR DESCRIPTION
**Pull request by solivier**
| Q | A |
| --- | --- |
| Review and 2 GTM | y |

### CODE REVIEW

File: features/Context/Page/Base/Base.php [hunk 1]
```diff
@@ -111,8 +111,14 @@ public function simpleFillField($locator, $value)
     public function toggleSwitch($locator, $on = true)
     {
         $field = $this->findField($locator);
+        $parent = $field->getParent();
+
         if ($field->isChecked() != $on) {
-            $field->getParent()->find('css', 'label')->click();
+            $parent = $this->spin(function () use ($field, $parent) {
+                return $parent->find('css', 'label');
+            }, 'Could not find parent element');
```
Comment (skeleton) [inline features/Context/Page/Base/Base.php]: are you sure about the message ? should not be "Could not find label element in parent" ?

File: features/Context/Page/Product/Edit.php [hunk 2]
```diff
@@ -800,13 +797,9 @@ public function clickCategoryFilterLink($category)
...
-        $node = $this
-            ->getCategoryTree()
-            ->find('css', '#node_-1 a');
-
-        if (null === $node) {
-            throw new \Exception(sprintf('Could not find unclassified category filter.'));
-        }
+        $node = $this->spin(function () {
+            return $this->getCategoryTree()->find('css', '#node_-1 a');
+        }, sprintf('Could not find unclassified category filter.'));
```
Comment (skeleton) [inline features/Context/Page/Product/Edit.php]: sprintf is useless

**Comment by skeleton** [comment]
gtm once fixed

File: features/Context/Page/Product/Edit.php [hunk 3]
```diff
@@ -784,13 +784,10 @@ public function selectTree($category)
      */
     public function clickCategoryFilterLink($category)
```
Comment (willy-ahva) [inline features/Context/Page/Product/Edit.php]: remove @throws pls

**Comment by willy-ahva** [comment]
GTM once fixed

File: features/Context/Page/Base/Base.php [hunk 4]
```diff
@@ -111,8 +111,14 @@ public function simpleFillField($locator, $value)
     public function toggleSwitch($locator, $on = true)
     {
         $field = $this->findField($locator);
+        $parent = $field->getParent();
+
         if ($field->isChecked() != $on) {
-            $field->getParent()->find('css', 'label')->click();
+            $parent = $this->spin(function () use ($field, $parent) {
```
Comment (jmleroux) [inline features/Context/Page/Base/Base.php]: Shouldn't it be `$parentLabel` ?

**Comment by jmleroux** [comment]
GTM once answered

**Pull request closed without merging**